astropy/units/tests/test_structured.py

```diff
@@ -625,6 +625,18 @@ def test_unstructured_to_structured(self):
         # For the other tests of ``structured_to_unstructured``, see
         # ``test_quantity_non_ufuncs.TestRecFunctions.test_unstructured_to_structured``
 
+    def test_merge_arrays(self):
```

Comment:
Maybe put this in `test_quantity_non_ufuncs` as well?

Reply:
✅